### DISCUSSION TITLE
Inferenced segmentation.dcm cannot be imported to Eclipse

### DISCUSSION THREAD

**Discussion by furtheraway** [microsoft/InnerEye-DeepLearning#687] [Q&A] [answered]
Hi all,

I am walking through the model taining and inference with the example lung segamentation dataset https://github.com/microsoft/InnerEye-DeepLearning/blob/main/docs/sample_tasks.md

The model training has succeed on Azure ML.

Then I zip up the DICOM files of the second test subject into a zip file:
C:\Users\xxxx\Downloads\IE_lung_dataset\manifest-1557326747206\LCTSC\LCTSC-Test-S1-102\11-04-2003-NA-RTRCCTTHORAX8FHigh Adult-20444\0.000000-CT114545RespCT  3.0  B30f  50 Ex-81163

`zip -r patS1102.zip 0.000000-CT114545RespCT\ \ 3.0\ \ B30f\ \ 50\ Ex-81163/`

Then submit it with submit_for_inference.py:

```
python InnerEye/Scripts/submit_for_inference.py --image_file /home/xxxx/datasets/inference_dicom/patS1102.zip --model_id LungExample:1 --download_folder /home/xxxx/source/2022/download/ --use_dicom true

```

The inference run finished successfully and I got the segmentation.dcm file. But when I tried to import this segmentation.dcm file to Eclipse, I run into the following error:

![image](https://user-images.githubusercontent.com/7104230/156857012-e7adb451-7660-45c4-89ab-92f93f6cfe17.png)

I can import the original DICOM files for this subject without any issue, 

@ant0nsc

**Comment by ant0nsc** [comment]
@furtheraway , thanks for reporting! 

I'm not familiar with Eclipse, could you please post a link to the software? If it's not open source, I see very little chance of finding out what went wrong there. The warning about transfer syntax is the only hint that we have at present. 

@fepegar, I think you have recently opened InnerEye results in another open source tool, was that slicer?

**Reply by fepegar** [reply]
Yes, I typically use 3D Slicer for visualisation. @furtheraway, can you please share the output DICOM file and, if possible, the corresponding input?

**Reply by furtheraway** [reply]
Thank you @fepegar  and @ant0nsc 

Here is the inference output segmentation.dcm.zip I got: (I noticed something interesting - the output file segmentation.dcm I got is always exactly 2048KB in size. I am not sure if this says something.)
[segmentation.dcm.zip](https://github.com/microsoft/InnerEye-DeepLearning/files/8229318/segmentation.dcm.zip)

The corresponding input is too big to attach in github. It is the first patient DICOM from the following public dataset:
https://wiki.cancerimagingarchive.net/display/Public/Lung+CT+Segmentation+Challenge+2017
I basically zip up the following folder in the above dataset as input for inference:
 ...\manifest-1557326747206\LCTSC\LCTSC-Test-S1-101\03-03-2004-NA-NA-08186\NA-79262

@fepegar I am not familiar with 3D slicer, and still trying to figure out how to display the inferenced segmentation result in it (any guide would be appreciated). But seems like at least the segmentation.dcm file can be loaded without error. Could you help me check if the output actually contain the countours we expect?

@ant0nsc Eclipse is the treatment planning system from Varian https://www.varian.com/products/radiotherapy/treatment-planning/eclipse
I don't think it's open source. haha

**Reply by furtheraway** [reply]
@fepegar  I realized how rediculous it is to ask you to download a 5GB dataset just to pull one patient from it. So I uploaded the input DICOM file to Google Drive, and shared it with the following link. Thank you : )

https://drive.google.com/file/d/16C4AhZnA6Tr8G5bqXBBt1PHhOEMaXagP/view?usp=sharing

**Reply by fepegar** [reply]
Thanks for sharing the data.

Slicer didn't work for me, which version are you using?

I opened the segmentation file with `pydicom`. It's a DICOM RT file. This is the output: https://we.tl/t-MHSgYmd0yC

Is that file all that was generated by InnerEye?

**Reply by furtheraway** [reply]
Thank you @fepegar 

I am using Slicer 4.11.20210226, and it doesn't run well on my computer at all. It always become inresponsive when I import or load DICOM data, even when I load the original DICOM input. So I am not sure if I loaded the segmentation.dcm properly. 

But I do saw, after force quiting and restarting Slicer (or waiting patiently after a while for unresponsive Slicer), the AutoSS series (as circled in the below screenshot). Its name suggests it is the segmentation results, but I don't know how to display it yet.

![image](https://user-images.githubusercontent.com/7104230/157926391-4935b48f-9808-45ce-bf22-4e7499831729.png)

Yes, segmentation.dcm.zip is the only file generated and returned by InnerEye inference. And its size is always 2048KB, even though I tried on two different subjects.

I don't know what to do with the pydicom.txt file you returned to me. I will look more into pydicom.

**Comment by furtheraway** [comment, marked as answer]
This issue has been addressed in version 0.7 code. 
Now segmentation.dcm load into Eclipse smoothly.
Thank you!
